examples_tests/60.ClusteredRendering/main.cpp

```diff
@@ -1,4 +1,3831 @@
+			1u, &lightGridUpdated);
+#endif
+
+		scanner->dispatchHelper(
+			commandBuffer.get(),
+			scanPipeline->getLayout(),
+			scanPushConstants,
+			scanDispatchInfo,
+			asset::EPSF_TOP_OF_PIPE_BIT,
+			0u, nullptr,
+			asset::EPSF_BOTTOM_OF_PIPE_BIT,
+			0u, nullptr);
+
+		// memory dependency to ensure the final culling pass has finished writing intersection records to the one of the scratches
+#ifdef SYNC_DEBUG
+		commandBuffer->pipelineBarrier(
+			asset::EPSF_ALL_COMMANDS_BIT,
+			asset::EPSF_ALL_COMMANDS_BIT,
+			asset::EDF_NONE,
+			1u, &debugSerializeAllBarrier,
+			0u, nullptr,
+			0u, nullptr);
+#else
+		cullScratchUpdated[0].buffer = octreeScratchBuffers[1];
+		commandBuffer->pipelineBarrier(
+			asset::EPSF_COMPUTE_SHADER_BIT,
+			asset::EPSF_COMPUTE_SHADER_BIT,
+			asset::EDF_NONE,
+			0u, nullptr,
+			1u, &cullScratchUpdated[0],
+			0u, nullptr);
+#endif
+
+#ifdef SYNC_DEBUG
+		commandBuffer->pipelineBarrier(
+			asset::EPSF_ALL_COMMANDS_BIT,
+			asset::EPSF_ALL_COMMANDS_BIT,
+			asset::EDF_NONE,
+			1u, &debugSerializeAllBarrier,
+			0u, nullptr,
+			0u, nullptr);
+#else
+		// image memory dependency to ensure that scan has finished writing to the light grid
+		lightGridUpdated.barrier.srcAccessMask = static_cast<asset::E_ACCESS_FLAGS>(asset::EAF_SHADER_READ_BIT | asset::EAF_SHADER_WRITE_BIT);
+		lightGridUpdated.barrier.dstAccessMask = asset::EAF_SHADER_READ_BIT;
+		commandBuffer->pipelineBarrier(
+			asset::EPSF_COMPUTE_SHADER_BIT,
+			asset::EPSF_COMPUTE_SHADER_BIT,
+			asset::EDF_NONE,
+			0u, nullptr,
+			0u, nullptr,
+			1u, &lightGridUpdated);
+#endif
+
+		commandBuffer->bindComputePipeline(scatterPipeline.get());
+		commandBuffer->bindDescriptorSets(asset::EPBP_COMPUTE, scatterPipeline->getLayout(), 0u, 1u, &scatterDS.get());
+		{
+			constexpr uint32_t MAX_INVOCATIONS = MEMORY_BUDGET / sizeof(uint64_t);
+			commandBuffer->dispatch((MAX_INVOCATIONS + WG_DIM - 1) / WG_DIM, 1u, 1u);
+		}
+
+		// memory dependency to ensure the light index list is updated
+		video::IGPUCommandBuffer::SBufferMemoryBarrier lightIndexListUpdated = {};
+		lightIndexListUpdated.barrier.srcAccessMask = asset::EAF_SHADER_WRITE_BIT;
+		lightIndexListUpdated.barrier.dstAccessMask = asset::EAF_SHADER_READ_BIT;
+		lightIndexListUpdated.srcQueueFamilyIndex = ~0u;
+		lightIndexListUpdated.dstQueueFamilyIndex = ~0u;
+		lightIndexListUpdated.buffer = lightIndexListGPUBuffer;
+		lightIndexListUpdated.offset = 0ull;
+		lightIndexListUpdated.size = lightIndexListUpdated.buffer->getCachedCreationParams().declaredSize;
+#ifdef SYNC_DEBUG
+		commandBuffer->pipelineBarrier(
+			asset::EPSF_ALL_COMMANDS_BIT,
+			asset::EPSF_ALL_COMMANDS_BIT,
+			asset::EDF_NONE,
+			1u, &debugSerializeAllBarrier,
+			0u, nullptr,
+			0u, nullptr);
+#else
+		commandBuffer->pipelineBarrier(
+			asset::EPSF_COMPUTE_SHADER_BIT,
+			asset::EPSF_FRAGMENT_SHADER_BIT,
+			asset::EDF_BY_REGION_BIT,
+			0u, nullptr,
+			1u, &lightIndexListUpdated,
+			0u, nullptr);
+#endif
+
+#endif
```

Comment:
how much of this can you kick to a `nbl::ext::ClusteredLighting::CClusteredLighting` class?